### PR TITLE
ExtraBeesEffect-WITHER - implementation.

### PR DESCRIPTION
**Pull request by RusTit**

### CODE REVIEW

File: extrabees/src/main/java/binnie/extrabees/genetics/effect/ExtraBeesEffect.java [hunk 1]
```diff
@@ -167,24 +162,14 @@ protected IEffectData doEffect(IBeeGenome genome, IEffectData storedData, IBeeHo
 	BLINDNESS {
 		@Override
 		protected IEffectData doEffect(IBeeGenome genome, IEffectData storedData, IBeeHousing housing, World world, BlockPos position) {
-			for (EntityPlayer player : this.getEntities(EntityPlayer.class, genome, housing)) {
-				if (world.rand.nextInt(4) < wearsItems(player)) {
-					continue;
-				}
-				player.addPotionEffect(new PotionEffect(MobEffects.BLINDNESS, 200));
-			}
+            addPotionEffect(genome, housing, world, MobEffects.BLINDNESS);
```
Comment (mezz) [inline extrabees/src/main/java/binnie/extrabees/genetics/effect/ExtraBeesEffect.java]: Weird indentation here

File: extrabees/src/main/java/binnie/extrabees/genetics/effect/ExtraBeesEffect.java [hunk 2]
```diff
@@ -167,24 +162,14 @@ protected IEffectData doEffect(IBeeGenome genome, IEffectData storedData, IBeeHo
...
 	CONFUSION{
 		@Override
 		protected IEffectData doEffect(IBeeGenome genome, IEffectData storedData, IBeeHousing housing, World world, BlockPos position) {
-			for (EntityPlayer player : this.getEntities(EntityPlayer.class, genome, housing)) {
-				if (world.rand.nextInt(4) < wearsItems(player)) {
-					continue;
-				}
-				player.addPotionEffect(new PotionEffect(MobEffects.NAUSEA, 200));
-			}
+            addPotionEffect(genome, housing, world, MobEffects.NAUSEA);
```
Comment (mezz) [inline extrabees/src/main/java/binnie/extrabees/genetics/effect/ExtraBeesEffect.java]: Weird indentation here

File: extrabees/src/main/java/binnie/extrabees/genetics/effect/ExtraBeesEffect.java [hunk 3]
```diff
@@ -313,7 +298,7 @@ protected IEffectData doEffect(IBeeGenome genome, IEffectData storedData, IBeeHo
 	WITHER{
 		@Override
 		protected IEffectData doEffect(IBeeGenome genome, IEffectData storedData, IBeeHousing housing, World world, BlockPos position) {
-			//TODO: add function ?
+            addPotionEffect(genome, housing, world, MobEffects.WITHER);
```
Comment (mezz) [inline extrabees/src/main/java/binnie/extrabees/genetics/effect/ExtraBeesEffect.java]: Weird indentation here

File: extrabees/src/main/java/binnie/extrabees/genetics/effect/ExtraBeesEffect.java [hunk 4]
```diff
@@ -334,15 +319,10 @@ protected IEffectData doEffect(IBeeGenome genome, IEffectData storedData, IBeeHo
...
 		@Override
 		protected IEffectData doEffect(IBeeGenome genome, IEffectData storedData, IBeeHousing housing, World world, BlockPos position) {
-			for (EntityPlayer player : this.getEntities(EntityPlayer.class, genome, housing)) {
-				if (world.rand.nextInt(4) < wearsItems(player)) {
-					continue;
-				}
-				player.addPotionEffect(new PotionEffect(MobEffects.WEAKNESS, 200));
-			}
-			return storedData;
+            addPotionEffect(genome, housing, world, MobEffects.WEAKNESS);
```
Comment (mezz) [inline extrabees/src/main/java/binnie/extrabees/genetics/effect/ExtraBeesEffect.java]: Weird indentation here

File: extrabees/src/main/java/binnie/extrabees/genetics/effect/ExtraBeesEffect.java [hunk 5]
```diff
@@ -334,15 +319,10 @@ protected IEffectData doEffect(IBeeGenome genome, IEffectData storedData, IBeeHo
...
+            return storedData;
 		}
-	},
+    },
```
Comment (mezz) [inline extrabees/src/main/java/binnie/extrabees/genetics/effect/ExtraBeesEffect.java]: Weird indentation here

File: extrabees/src/main/java/binnie/extrabees/genetics/effect/ExtraBeesEffect.java [hunk 6]
```diff
@@ -397,6 +377,27 @@ protected IEffectData doEffect(IBeeGenome genome, IEffectData storedData, IBeeHo
 		}
 	};
 
+	private final static int DEFAULT_POTION_DURATION = 200;
+
+    protected void addPotionEffect(final IBeeGenome genome, final IBeeHousing housing, final World world, final Potion effect) {
```
Comment (mezz) [inline extrabees/src/main/java/binnie/extrabees/genetics/effect/ExtraBeesEffect.java]: Lots more weird indentation here

**Review by mezz — CHANGES_REQUESTED** [review]
Looks good, just has some formatting issues with indentation

**Comment by RusTit** [comment]
Can I do another auto-intend to all file?

**Comment by mezz** [comment]
Feel free, yeah. As long as the auto-formatting is a separate commit it's still easy to review.

**Comment by RusTit** [comment]
Done

**Pull request merged**